### PR TITLE
Fix for CVE-2024-39908 in rubygem-rexml

### PR DESCRIPTION
**Pull request by bhagyapathak**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ *] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ *] The toolchain/worker package manifests are up-to-date
- [ *] Any updated packages successfully build (or no packages were changed)
- [ *] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ *] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ *] All package sources are available
- [ *] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ *] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [ *] All source files have up-to-date hashes in the `*.signatures.json` files
- [ *] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ *] Documentation has been updated to match any changes to the build system
- [ *] Ready to merge

---

###### Summary 

Fix for CVE-2024-39908

###### Change Log  

Upgrade rubygem-rexml to 3.3.4

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2024-39908

###### Test Methodology

- Pipeline build id: xxxx

### CODE REVIEW

File: SPECS/rubygem-rexml/rubygem-rexml.spec [hunk 1]
```diff
@@ -2,7 +2,7 @@
 %global gem_name rexml
 Summary:        REXML is an XML toolkit for Ruby
 Name:           rubygem-%{gem_name}
-Version:        3.2.7
+Version:        3.3.4
```
Comment (mfrw) [inline SPECS/rubygem-rexml/rubygem-rexml.spec:5]: The CVE seems to be a medium. Should we target the `main` branch ?

**Review by PawelWMS — CHANGES_REQUESTED** [review]
+1 to @mfrw's comment - why push the fix through fast-track if it's a medium? For such cases going to `main` is usually good enough. Unless we're over of very close to our SLA for this one.

File: SPECS/rubygem-rexml/rubygem-rexml.spec (hunk 1 above)
Comment (bhagyapathak) [reply SPECS/rubygem-rexml/rubygem-rexml.spec:5]: I am closing this PR, will open new PR for main branch.

Comment (bhagyapathak) [reply SPECS/rubygem-rexml/rubygem-rexml.spec:5]: New PR - https://github.com/microsoft/azurelinux/pull/10187

**Pull request closed without merging**